src/openshift/service.ts

```diff
@@ -8,38 +8,79 @@ import { OpenShiftObject, Command } from '../odo';
+            });
+            if (!serviceTemplateName) return null;
 
-        const serviceTemplatePlanName = await vscode.window.showQuickPick(Service.odo.getServiceTemplatePlans(serviceTemplateName), {
-            placeHolder: "Service Template Plan Name"
-        });
-        if (!serviceTemplatePlanName) return null;
+            const serviceTemplatePlanName = await vscode.window.showQuickPick(Service.odo.getServiceTemplatePlans(serviceTemplateName), {
+                placeHolder: "Service Template Plan Name"
+            });
+            if (!serviceTemplatePlanName) return null;
 
-        const serviceName = await vscode.window.showInputBox({
-            value: serviceTemplateName,
-            prompt: 'Service Name',
-            validateInput: (value: string) => {
-                // required, because dc name is ${component}-${app}
-                let message: string = null;
-                if (`${value.trim()}-${application.getName()}`.length > 63) {
-                    message = 'Service name cannot be more than 63 characters';
+            const serviceName = await vscode.window.showInputBox({
+                value: serviceTemplateName,
+                prompt: 'Service Name',
+                validateInput: (value: string) => {
+                    // required, because dc name is ${component}-${app}
+                    let message: string = null;
+                    if (`${value.trim()}-${applicationName.getName()}`.length > 63) {
+                        message = 'Service name cannot be more than 63 characters';
+                    }
+                    return message;
                 }
-                return message;
+            });
+            if (serviceName) {
+                const project = applicationName.getParent();
+                return Progress.execCmdWithProgress(`Creating new service '${serviceName}'`,
+                    Command.createService(project.getName(), applicationName.getName(), serviceTemplateName, serviceTemplatePlanName, serviceName.trim()))
+                    .then(() => Service.explorer.refresh())
+                    .then(() => `Service '${serviceName}' successfully created`)
+                    .catch((err) => Promise.reject(`Failed to create service with error '${err}'`));
             }
-        });
-        if (serviceName) {
-            const project = application.getParent();
-            return Progress.execCmdWithProgress(`Creating new service '${serviceName}'`,
-                Command.createService(project.getName(), application.getName(), serviceTemplateName, serviceTemplatePlanName, serviceName.trim()))
-                .then(() => Service.explorer.refresh(application))
-                .then(() => `Service '${serviceName}' successfully created`)
-                .catch((err) => Promise.reject(`Failed to create service with error '${err}'`));
+            return null;
         }
-        return null;
+    }
+
+    static async getApplicationData(application) {
```

Comment:
Please see https://github.com/redhat-developer/vscode-openshift-tools/pull/432#discussion_r245595571 and #499 for the summary of my issues with this implementation.